nav2_rrtconnect_planner/src/rrt_connect.cpp

```diff
@@ -0,0 +1,269 @@
+
+#include <iostream>
+#include <fstream>
+#include <utility>
+
+#include "nav2_util/node_utils.hpp"
+#include "nav2_rrtconnect_planner/rrt_connect.hpp"
+
+namespace nav2_rrtconnect_planner
+{
+
+RRTConnect::RRTConnect()
+: tf_(nullptr), costmap_(nullptr), ss_(nullptr), ompl_state_space_(nullptr), bounds_(nullptr)
+{
+
+}
+
+void RRTConnect::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros)
+{
+
+  node_ = parent;
+  tf_ = tf;
+  name_ = name;
+  costmap_ = costmap_ros->getCostmap();
+  global_frame_ = costmap_ros->getGlobalFrameID();
+
+  ompl_initialized_ = false;
+
+  RCLCPP_INFO(
+    node_->get_logger(), "Configuring plugin %s of type NavfnPlanner",
+    name_.c_str());
+
+  // Parameter initialization
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".solve_time",
+    rclcpp::ParameterValue(1.0));
+  node_->get_parameter(name_ + ".solve_time", solve_time_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".max_trials",
+    rclcpp::ParameterValue(4));
+  node_->get_parameter(name_ + ".max_trials", max_trials_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".collision_checking_resolution", rclcpp::ParameterValue(
+      0.001));
+  node_->get_parameter(name_ + ".collision_checking_resolution", collision_checking_resolution_);
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".allow_unknown", rclcpp::ParameterValue(
+      true));
+  node_->get_parameter(name_ + ".allow_unknown", allow_unknown_);
+}
+
+void RRTConnect::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type NavfnPlanner",
+    name_.c_str());
+}
+
+void RRTConnect::initializeOMPL()
+{
+  // Defining Bounds
+  bounds_ = new ompl::base::RealVectorBounds(2);
+  bounds_->setLow(0, costmap_->getOriginX());
+  bounds_->setHigh(
+    0, costmap_->getSizeInMetersX() +
+    costmap_->getOriginX());
+  bounds_->setLow(1, costmap_->getOriginY());
+  bounds_->setHigh(
+    1, costmap_->getSizeInMetersY() +
+    costmap_->getOriginY());
+
+  RCLCPP_INFO(
+    node_->get_logger(),
+    "bounds: %f %f %f %f", bounds_->low[0], bounds_->low[1],
+    bounds_->high[0], bounds_->high[1]);
+
+  ompl_state_space_ = std::make_shared<ompl::base::SE2StateSpace>();
+  // Bounding the R2 state space
+  ompl_state_space_->as<ompl::base::SE2StateSpace>()->setBounds(*bounds_);
+
+  // Simple setup
+  ss_.reset(new ompl::geometric::SimpleSetup(ompl_state_space_));
+  // Setting state validity checker
+  ss_->setStateValidityChecker(
```

Comment:
Oh this is odd, why is this a lambda? Does a typical std::bind not work?

Reply:
std::bind does work but I thought lambda would be more efficient. what do you think?

Reply:
std::bind doesn't require creating a new lambda to just call the same function - we should just use that.

Reply:
Scott Meyers in his book Effective Modern C++ makes a good case for using lambda exprs instead of std::bind. See https://github.com/peter-can-write/cpp-notes/blob/master/prefer-lambdas-over-std::bind.md

Reply:
I disagree with the initial premise `Lambdas are more readable and clear...` but I see the point. This is fine as is.